test/hikari_cp/core_test.clj

```diff
@@ -55,14 +55,18 @@
 
 (def mysql-datasouurce-config
   (datasource-config (merge valid-options
-                            {:adapter "mysql" :use-legacy-datetime-code false})))
+                            {:adapter "mysql"
+                             :datasource-classname "com.mysql.cj.jdbc.MysqlDataSource"
+                             :use-legacy-datetime-code false})))
 
 (def metric-registry-config (datasource-config (merge valid-options metric-registry-options)))
 
 (def health-check-registry-config (datasource-config (merge valid-options health-check-registry-options)))
 
 (expect false
         (get (.getDataSourceProperties mysql-datasouurce-config) "useLegacyDatetimeCode"))
+(expect "com.mysql.cj.jdbc.MysqlDataSource"
```

Comment:
Just to be sure @vigneshsarma - this example fails without your fix? :)

Reply:
😄 Yes, I tested it against `master`.

### `master`
```
➜  hikari-cp git:(master) ✗ lein test
failure in (core_test.clj:68) : hikari-cp.core-test
(expect "foobar" (.getDataSourceClassName mysql-datasouurce-config))

           expected: "foobar"
                was: "com.mysql.jdbc.jdbc2.optional.MysqlDataSource"

           matches: ""
           diverges: "foobar"
                  &: "com.mysql.jdbc.jdbc2.optional.MysqlDataSource"

Ran 68 tests containing 68 assertions in 2147 msecs
1 failures, 0 errors.
```
### `allow-custom-datasource`
```
➜  hikari-cp git:(allow-custom-datasource) ✗ lein test

Ran 68 tests containing 68 assertions in 1156 msecs
0 failures, 0 errors.
```